plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestTransactionLogAccess.java

```diff
@@ -736,6 +737,63 @@ public void testTableSnapshotsCacheDisabled()
                         "00000000000000000014.json", 2));
     }
 
+    @Test
+    public void testTableSnapshotsActiveDataFilesCache()
+            throws Exception
+    {
+        String tableName = "person";
+        Path tableDir = new Path(getClass().getClassLoader().getResource("databricks/" + tableName).toURI());
+        DeltaLakeConfig shortLivedActiveDataFilesCacheConfig = new DeltaLakeConfig();
+        shortLivedActiveDataFilesCacheConfig.setDataFileCacheTtl(new Duration(10, TimeUnit.MINUTES));
+        setupTransactionLogAccess(tableName, tableDir, shortLivedActiveDataFilesCacheConfig);
+
+        List<AddFileEntry> addFileEntries = transactionLogAccess.getActiveFiles(tableSnapshot, SESSION);
+        assertEquals(addFileEntries.size(), 12);
+        assertEquals(
+                hdfsEnvironment.getAccessedPathNames(),
+                ImmutableMap.of(
+                        "person", 1,
+                        "00000000000000000010.checkpoint.parquet", 2));
+
+        addFileEntries = transactionLogAccess.getActiveFiles(tableSnapshot, SESSION);
+        assertEquals(addFileEntries.size(), 12);
+        // The internal data cache should still contain the data files for the table
+        assertEquals(
+                hdfsEnvironment.getAccessedPathNames(),
+                ImmutableMap.of(
+                        "person", 1,
+                        "00000000000000000010.checkpoint.parquet", 2));
+    }
+
+    @Test
+    public void testTableSnapshotsActiveDataFilesCacheDisabled()
+            throws Exception
+    {
+        String tableName = "person";
+        Path tableDir = new Path(getClass().getClassLoader().getResource("databricks/" + tableName).toURI());
+        DeltaLakeConfig shortLivedActiveDataFilesCacheConfig = new DeltaLakeConfig();
+        shortLivedActiveDataFilesCacheConfig.setDataFileCacheTtl(new Duration(0, TimeUnit.SECONDS));
+        setupTransactionLogAccess(tableName, tableDir, shortLivedActiveDataFilesCacheConfig);
+
+        List<AddFileEntry> addFileEntries = transactionLogAccess.getActiveFiles(tableSnapshot, SESSION);
+        assertEquals(addFileEntries.size(), 12);
+        assertEquals(
+                hdfsEnvironment.getAccessedPathNames(),
+                ImmutableMap.of(
+                        "person", 1,
+                        "00000000000000000010.checkpoint.parquet", 2));
```

Comment:
I'm trying to find out whether we can avoid reading the checkpoint file twice when retrieving the active files.